pennylane/templates/layers/basic_entangler.py

```diff
@@ -157,14 +122,91 @@ def circuit(weights):
         ``ValueError: Wrong number of parameters``.
     """
 
-    if rotation is None:
-        rotation = RX
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, weights, wires=None, rotation=None, do_queue=True):
+
+        self.rotation = rotation or qml.RX
+
+        super().__init__(weights, wires=wires, do_queue=do_queue)
+        self._preprocess()
+
+    def expand(self):
+
+        weights = self.data[0]
+
+        # first dimension of the weights tensor determines
+        # the number of layers
+        repeat = qml.math.shape(weights)[0]
+
+        with qml.tape.QuantumTape() as tape:
+
+            for layer in range(repeat):
+                for i in range(len(self.wires)):
+                    self.rotation(weights[layer][i], wires=self.wires[i : i + 1])
+
+                broadcast(unitary=qml.CNOT, pattern="ring", wires=self.wires)
+
+        return tape
+
+    def _preprocess(self):
+        """Validate and pre-process inputs as follows:
+
+        * Check the shape of the weights tensor, making sure that the second dimension
+          has length :math:`n`, where :math:`n` is the number of qubits.
+        """
+        shape = qml.math.shape(self.parameters[0])
+
+        if len(shape) != 2:
+            raise ValueError(f"Weights tensor must be 2-dimensional; got shape {shape}")
+
+        if shape[1] != len(self.wires):
+            raise ValueError(
+                f"Weights tensor must have second dimension of length {len(self.wires)}; got {shape[1]}"
+            )
+
+    @staticmethod
```

Comment:
A static method does not take `self` as the first argument, which means it does not have access to properties of the class like `self.parameters`. The advantage is that one can call it without making an instance of the class, i.e., one can just do

```
from pennyLane.templates import BasicEntanglerLayers

BasicEntanglerLayers.weights_normal()
```
instead of having to do

```
from pennyLane.templates import BasicEntanglerLayers

my_op = BasicEntanglerLayers(...) 
my_op.weights_normal()
```

Reply:
In some sense, static methods just know about the general class, they do not know what arguments an instance was created with. Hence the need to feed `n_layers` etc into this function. 

The reason to use this here (although an initialisation that knows the number of layers etc would be great) is that ``BasicEntanglerLayers(...) `` would be typically created in a qfunc, but the user needs the weights outside of a qfunc.